src/test/scala/org/camunda/feel/impl/builtin/BuiltinListFunctionsTest.scala

```diff
@@ -570,4 +570,17 @@ class BuiltinListFunctionsTest
     evaluateExpression(" is empty([1,2,3]) ") should returnResult(false)
     evaluateExpression(" is empty(list: [1]) ") should returnResult(false)
   }
+
+  "A partition() function" should "return list partitioned by _" in {
+    evaluateExpression(" partition([1,2,3,4,5], 2) ") should returnResult(
+      List(List(1, 2), List(3, 4), List(5))
+    )
+    evaluateExpression(" partition([], 2) ") should returnResult(List())
+    evaluateExpression(" partition([1], 2) ") should returnResult(List(List(1)))
+  }
+
+  it should "return null if the size is invalid" in {
+    evaluateExpression(" partition([1,2], 0) ") should returnNull()
```

Comment:
:wrench: After returning an error from the function, we could verify the error in the test case.

```
    evaluateExpression(" partition([1,2], 0) ") should (
      returnNull() and reportFailure(
        FUNCTION_INVOCATION_FAILURE,
        "Failed to invoke function 'partition': 'size' should be greater than zero but was '0'"
      )
    )
```